multimap/src/test/java/org/infinispan/multimap/impl/DistributedMultimapCacheTest.java

```diff
@@ -0,0 +1,91 @@
+package org.infinispan.multimap.impl;
+
+import static java.lang.String.format;
+import static org.infinispan.functional.FunctionalTestUtils.await;
+import static org.testng.AssertJUnit.assertEquals;
+import static org.testng.AssertJUnit.assertNotNull;
+import static org.testng.AssertJUnit.assertTrue;
+
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.infinispan.Cache;
+import org.infinispan.container.DataContainer;
+import org.infinispan.container.entries.ImmortalCacheEntry;
+import org.infinispan.container.entries.InternalCacheEntry;
+import org.infinispan.distribution.BaseDistFunctionalTest;
+import org.infinispan.manager.EmbeddedCacheManager;
+import org.infinispan.multimap.EmbeddedMultimapCacheManagerFactory;
+import org.infinispan.multimap.api.MultimapCache;
+import org.infinispan.test.TestingUtil;
+import org.testng.annotations.Test;
+
+@Test(groups = "functional", testName = "distribution.DistributedMultimapCacheTest")
+public class DistributedMultimapCacheTest extends BaseDistFunctionalTest<String, Collection<String>> {
+
+   private Map<String, MultimapCache<String, String>> multimapCacheCluster = new HashMap<>();
+
+   @Override
+   protected void createCacheManagers() throws Throwable {
+      super.createCacheManagers();
+
+      for (EmbeddedCacheManager cacheManager : cacheManagers) {
+         multimapCacheCluster.put(cacheManager.getAddress().toString(),
+               EmbeddedMultimapCacheManagerFactory.get(cacheName, cacheManager, configuration.build()));
+      }
+   }
+
+   @Override
+   protected void initAndTest() {
+      for (MultimapCache<String, String> mc : multimapCacheCluster.values()) {
+         await(mc.size().thenAccept(s -> assertEquals(0, s.intValue())));
```

Comment:
I created an assertion for the size that I will share between tests. Distributed test is in very WIP because it fails with eval